src/main/java/com/iota/iri/IRI.java

```diff
@@ -361,9 +118,49 @@ private static void shutdownHook() {
                 }
             }, "Shutdown Hook"));
         }
-    }
 
-    public static boolean validateParams(Configuration configuration, String[] args) throws IOException {
-        return IRILauncher.isValidated(configuration, args);
+        private static IotaConfig createConfiguration(String[] args) {
+            IotaConfig iotaConfig = null;
+            boolean testnet = ArrayUtils.contains(args, Config.TESTNET_FLAG);
+            File configFile = chooseConfigFile(args);
+            String message = "Configuration is created using ";
+
+            try {
+                if (configFile.exists()) {
+                    iotaConfig = ConfigFactory.createFromFile(configFile, testnet);
+                    message+= configFile.getName() + " and command line args";
+                }
+                else {
+                    iotaConfig = ConfigFactory.createIotaConfig(testnet);
+                    message+= "command line args only";
+                }
+                JCommander jCommander = iotaConfig.parseConfigFromArgs(args);
+                if (iotaConfig.isHelp()) {
+                    jCommander.usage();
+                    System.exit(0);
+                }
+            }
+            catch (IOException | IllegalArgumentException e) {
+                log.error("There was a problem reading configuration from file" , e.getMessage());
+                log.debug("" ,e);
+                System.exit(-1);
+            }
+            catch (ParameterException e) {
+                log.error("There was a problem parsing commandline arguments: {}", e.getMessage());
+                log.debug("", e);
+                System.exit(-1);
+            }
+            log.info(message);
+            log.info("parsed the following cmd args: {}", Arrays.toString(args));
+            return iotaConfig;
+        }
+
+        private static File chooseConfigFile(String[] args) {
+            int index = Math.max(ArrayUtils.indexOf(args, "-c"), ArrayUtils.indexOf(args, "--config"));
+            if (index == -1) {
+                return IotaConfig.CONFIG_FILE;
+            }
+            return new File(args[++index]);
```

Comment:
this throws an `ArrayIndexOutOfBounds` exception if the file name is missing - not so clear
```
$ java -jar iri-1.5.3.jar -c
Logging - property 'logging-level' set to: [INFO]
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 1
	at com.iota.iri.IRI$IRILauncher.chooseConfigFile(IRI.java:163)
	at com.iota.iri.IRI$IRILauncher.createConfiguration(IRI.java:125)
	at com.iota.iri.IRI$IRILauncher.main(IRI.java:68)
	at com.iota.iri.IRI.main(IRI.java:31)
```